build-system/eslint-rules/closure-type-primitives.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const doctrine = require('doctrine');
+const traverse = require('traverse');
+
+/** @typedef {!Object} */
+let EslintContextDef;
+
+/** @typedef {!Object} */
+let EslintNodeDef;
+
+/**
+ * @typedef {{
+ *   node: !EslintNodeDef,
+ *   parsed: !Object
+ * }}
+ */
+let ClosureCommentDef;
+
+module.exports = function(context) {
+  const sourceCode = context.getSourceCode();
+
+  return {
+    meta: {
+      fixable: 'code',
+    },
+    Program: function() {
+      const comments =
+          /** @type {!Array<!EslintNodeDef>} */ (sourceCode.getAllComments());
+      comments
+          .map(node => parseClosureComments(context, node))
+          .filter(Boolean)
```

Comment:
took me a bit to realize what `.filter(Boolean)` does. Cool but... unreadable :) can `checkClosureComments` just check for falsy instead so we don't need to remove them from array?